Fix/cleanup aws workflow

rugeli opened the pull request:
Problem
=======
What is the problem this work solves, including
closes #304 

Solution
========
What I/we did to solve this problem
- Replace the `Install AWS CLI` step with a script that installs CLI directly from AWS
- Included the `--update` flag to handle the existing installations

  
## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)

Steps to Verify:
----------------
1. This workflow should be triggered by closing a pull request 
2. I have tested this workflow on a testing branch, and AWS CLI was installed as expected

mogres commented:
Hi Ruge, the updates looks good! However, it looks like AWS CLI is preinstalled on github hoster runners as seen [here](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md) in the CLI tools section (more documentation [here](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners)). Can we check on the test branch whether removing the setup step has any effect? If so I would recommend just getting rid of the install step and using the preconfigured version of AWS CLI.

rugeli commented:
> Hi Ruge, the updates look good! However, it looks like AWS CLI is preinstalled on github hosted runners as seen [here](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md) in the CLI tools section (more documentation [here](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners)). Can we check on the test branch whether removing the setup step has any effect? If so I would recommend just getting rid of the install step and using the preconfigured version of AWS CLI.

Good point! We can simply rely on the preinstalled version. I've removed the installation step and added a step to check the version instead. It worked fine. Thank you for catching that!

mogres reviewed (approved):
Looks good! I like the idea of not having to download packages through `apt-get` as that can be sometimes unreliable
